cvat/apps/engine/tests/utils.py

```diff
@@ -125,6 +128,252 @@ def setUp(self):
         super().setUp()
         self.client = self.client_class()
 
+    def _get_request(self, path: str, user: str, *, data: dict[str, Any] | None = None) -> Response:
+        with ForceLogin(user, self.client):
+            response = self.client.get(path, data=data)
+        return response
+
+    def _delete_request(self, path: str, user: str):
+        with ForceLogin(user, self.client):
+            response = self.client.delete(path)
+        return response
+
+    def _post_request(self, path: str, user: str, *, data: dict[str, Any] | None = None):
+        with ForceLogin(user, self.client):
+            response = self.client.post(path, data=data)
+        return response
+
+    def _put_request(self, url: str, user: str, *, data: dict[str, Any] | None = None):
+        with ForceLogin(user, self.client):
+            response = self.client.put(url, data=data)
+        return response
+
+    def _wait_request_to_be_finished(
```

Comment:
It's odd to see this here. IIRC, for unit tests we run RQ jobs in synchronous mode, so no waiting should be necessary - the job should be complete immediately after the initiating request.

Reply:
it's true, but I believe there are still several advantages of adding a waiting step:
1. We do not need to update related tests if we run these tests in asynchronous mode
2. This way we test that `/api/requests/rq_id` API returns the expected code (200, 401, 403)

Reply:
>  if we run these tests in asynchronous mode

Is it possible?

> This way we test that /api/requests/rq_id API returns the expected code (200, 401, 403)

I've nothing against calling this API, I just don't think we need to do it in a loop.

Reply:
> Warum nicht?

Well... how would you do it? AFAICS, `cvat.settings.testing` forces all queues to use sync mode. Moreover, it also uses a fake Redis implementation, so it wouldn't be possible to run RQ workers anyway.